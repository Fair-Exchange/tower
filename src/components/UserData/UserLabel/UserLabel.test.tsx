import {UserLabel, UserLabelProps} from './index';
import {shallow} from 'enzyme';
import * as React from 'react';

const defaults: UserLabelProps = {
    classes: {},
    user: {labels: []},
    newLabelName: '',
    newLabelValue: '',
    newLabelScope: '',
    addNewLabel: jest.fn(),
    editLabel: jest.fn(),
    deleteUserLabel: jest.fn(),
    changeLabelName: jest.fn(),
    changeLabelScope: jest.fn(),
    changeLabelValue: jest.fn(),
    isAddLabelModalOpened: false,
    isEditLabelModalOpened: false,
    openAddLabelModal: jest.fn(),
    openEditLabelModal: jest.fn(),
    closeModal: jest.fn(),
};

describe('UserLabel component', () => {
    const setup = (props: Partial<UserLabelProps> = {}) =>
        shallow(<UserLabel {...{...defaults, ...props}}/>);

    it('should render', () => {
        const wrapper = setup();
        expect(wrapper).toBeDefined();
    });
});
