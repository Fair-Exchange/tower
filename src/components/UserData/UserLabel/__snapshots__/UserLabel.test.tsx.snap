// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLabel component should render 1`] = `
<div>
  <WithStyles(Grid)
    item={true}
    style={
      Object {
        "display": "flex",
        "justifyContent": "space-between",
      }
    }
    xs={12}
  >
    <WithStyles(Typography)
      gutterBottom={true}
      style={
        Object {
          "paddingLeft": 0,
          "paddingTop": 0,
        }
      }
      variant="h6"
    >
      Labels
    </WithStyles(Typography)>
    <WithStyles(Grid)
      item={true}
    >
      <WithStyles(Button)
        onClick={[Function]}
      >
        <pure(AddCircleOutlineIcon)
          style={
            Object {
              "color": "#979797",
              "opacity": 0.6,
            }
          }
        />
      </WithStyles(Button)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
  <WithStyles(Typography)
    gutterBottom={true}
    style={
      Object {
        "color": "#757575",
      }
    }
    variant="h6"
  >
    No labels
  </WithStyles(Typography)>
</div>
`;
